### PR TITLE
example-log-extension: refactor code + update README

### PR DESCRIPTION
**Pull request by virkt25**
### Commit 1
- renamed LogLevelMixin to LogMixin
- made LogMixin automatically mount LogComponent for better UX
- added tests and refactored some as well.

### Commit 2
- Reviewed and updated code in README

connected to #988 

## Checklist

- [x] `npm test` passes on your machine
- [x] New tests added or existing tests modified to cover all changes
- [x] Code conforms with the [style guide](http://loopback.io/doc/en/contrib/style-guide.html)
- [ ] Related API Documentation was updated
- [ ] Affected artifact templates in `packages/cli` were updated
- [x] Affected example projects in `packages/example-*` were updated

### CODE REVIEW

File: packages/example-log-extension/README.md [hunk 1]
```diff
@@ -8,37 +8,38 @@ This repository shows you how to use [@loopback/cli](https://github.com/stronglo
...
-To use the extension, load the component to get access to a `LogFn` that can be
-used in a sequence to log information. A Mixin allows you to set the
-application wide logLevel. Only Controller methods configured at or above the
+To use this extension you can add the `LogMixin` to your Application which will
+provide you a function to set the Application wide log level as well as automatically
+load the `LogComponent`. Only Controller methods configured at or above the
 logLevel will be logged.
 
+_You may alternatively load `LogCompoent` yourself and set the log level using
+the appropriate binding keys manually if you don't wish to use the `LogMixin`.
```
Comment (shimks) [inline packages/example-log-extension/README.md]: Incomplete italics

File: packages/example-log-extension/test/unit/mixins/log.mixin.unit.ts [hunk 2]
```diff
@@ -5,32 +5,42 @@
...
 
-  function expectLogLevelToBeBound(myApp: Application) {
+  class AppWithLogMixin extends LogMixin(Application) {}
+
+  function expectLogLevelToBeBound() {
     const logLevel = myApp.getSync(EXAMPLE_LOG_BINDINGS.APP_LOG_LEVEL);
     expect(logLevel).to.be.eql(LOG_LEVEL.ERROR);
   }
+
+  function getApp() {
```
Comment (shimks) [inline packages/example-log-extension/test/unit/mixins/log.mixin.unit.ts]: Probably should be renamed to something like `setUpApp`

Comment (virkt25) [reply packages/example-log-extension/test/unit/mixins/log.mixin.unit.ts]: Most of out helper `before(Each)` hooks are `getApp` or `givenApp`.

File: packages/example-log-extension/README.md [hunk 3]
```diff
@@ -8,37 +8,38 @@ This repository shows you how to use [@loopback/cli](https://github.com/stronglo
...
 
-To use the extension, load the component to get access to a `LogFn` that can be
-used in a sequence to log information. A Mixin allows you to set the
-application wide logLevel. Only Controller methods configured at or above the
+To use this extension you can add the `LogMixin` to your Application which will
+provide you a function to set the Application wide log level as well as automatically
+load the `LogComponent`. Only Controller methods configured at or above the
 logLevel will be logged.
 
+_You may alternatively load `LogCompoent` yourself and set the log level using
```
Comment (b-admike) [inline packages/example-log-extension/README.md]: `LogCompoent` -> `LogComponent`

File: packages/example-log-extension/README.md [hunk 4]
```diff
@@ -109,8 +110,16 @@ export enum LOG_LEVEL {
 }
 ```
 
+
 ### `src/types.ts`
-Define TypeScript type definitions / interfaces for complex types and functions here.
+Before we continue, we will need to install a new dependecy as follows:
+
+```shell
+npm i @loopback/rest
```
Comment (b-admike) [inline packages/example-log-extension/README.md]: `--save` or `--save-dev` here?

Comment (virkt25) [reply packages/example-log-extension/README.md]: `--save` is the default behaviour with npm 5

File: packages/example-log-extension/README.md [hunk 5]
```diff
@@ -217,16 +226,18 @@ providing it via `ApplicationOptions` or using a helper method `app.logLevel(lev
 ```ts
 import {Constructor} from '@loopback/context';
 import {EXAMPLE_LOG_BINDINGS} from '../keys';
+import {LogComponent} form '../component';
```
Comment (b-admike) [inline packages/example-log-extension/README.md]: `form` -> `from`

**Pull request merged**